### DISCUSSION TITLE
Localization of create/edit/view pages

### DISCUSSION THREAD

**Discussion by chiiya** [filamentphp/filament#901] [Ideas]
I think there should be a better default behavior for localization of the page title for create/edit/view pages. Currently, if the `getTitle()` method is not manually overwritten, it falls back to the class name. I think it should instead fall back to use something like this, using the resource label:

```php
$resource = static::getResource()::getLabel();

return __('Edit :resource', ['resource' => $resource]);
```

Same for create/view. This would also be consistent with the index page, where something like this is being used to localize the "Create :resource" button label.

**Comment by danharrin** [comment]
Sure, that's a good idea. PR if you wish, we'll need a `title` property for resource page translation files :)

**Comment by chiiya** [comment]
Right. While investigating this, I noticed that every resource page has its own translation files, with a decent amount of duplication. Is there a specific reason you chose this over what e.g. Nova does with a single JSON file? I feel like while you lose a bit of flexibility, the latter is much easier to maintain, especially for community translations. As an example, some translations are already missing entries, even though the same strings have already been translated elsewhere.

IMO ideally translations should not be part of the core package. There should be something like [laravel-nova-lang](https://github.com/coderello/laravel-nova-lang) specialized for community-provided translations. The automated checks they have for missing entries and stuff is really useful.

**Reply by danharrin** [reply]
Hey, I definitely don't want to outsource translations to a separate project. I feel like it's important that they are in the core to allow non-English users to use the project out of the box without relying on other people.

The problem with out-of-sync translations is not automatically fixed by outsourcing, and I'd happily integrate tools that fix it into core codebase. I've thought about creating CLIs for this sort of thing before :)

As for duplication - I agree that there are several duplicate keys, but in many languages, context matters. Here's a good article on why it's better to be safe with context-based keys, rather than consuming the same keys in multiple places - https://www.languageintelligence.com/post/software-translation-challenges-the-big-three.

I'm gonna open a PR that adds a `$title` attribute to all pages to solve this issue. Thanks for your input as always :)

**Comment by danharrin** [comment]
Hey @chiiya, I've added the translatable titles in #934.

**Reply by chiiya** [reply]
Thanks!
